### ISSUE TITLE
Sleep instant wake when using Intel AX210 Bluetooth

### ISSUE THREAD

**Issue by deanls1** [perez987/macOS-14-Sonoma-on-z390-with-OpenCore#3]
I noticed you've mentioned this issue in two places before. Just wondering, have you managed to solve it yet?

<img width="1143" alt="image" src="https://github.com/user-attachments/assets/bc8120f0-5be0-465f-b51e-a9860b06de27">

https://github.com/OpenIntelWireless/IntelBluetoothFirmware/issues/477

**Comment by perez987** [comment]
@deanls1 
Yes, there is a fix although waking by USB mouse or keyboard is lost. Sleep is okay but system wakes only by power button. Read [this](https://www.insanelymac.com/forum/topic/358671-sleep-instant-wake-when-using-intel-ax210-bluetooth/?do=findComment&comment=2816737).

**Comment by deanls1** [comment]
> @deanls1 Yes, there is a fix although waking by USB mouse or keyboard is lost. Sleep is okay but system wakes only by power button. Read [this](https://www.insanelymac.com/forum/topic/358671-sleep-instant-wake-when-using-intel-ax210-bluetooth/?do=findComment&comment=2816737).

I'll try out the methods you mentioned. Thanks a lot!

**Comment by perez987** [comment]
@deanls1 
Other users have tried it and it has worked well. It is inconvenient to have to wake up the system using the power button, but at least the system goes to sleep mode and wakes up as it should.

**Comment by deanls1** [comment]
> @deanls1 Other users have tried it and it has worked well. It is inconvenient to have to wake up the system using the power button, but at least the system goes to sleep mode and wakes up as it should.

@perez987 I tried it, but it might not be the right fit for me. After using SSDT-S3, my machine restarts automatically after going to sleep. I’ll look for some SSDT-S3 files that might work for my setup and give it another shot. Thanks a lot for your help!

**Comment by deanls1** [comment]
> > @deanls1 Other users have tried it and it has worked well. It is inconvenient to have to wake up the system using the power button, but at least the system goes to sleep mode and wakes up as it should.
> 
> @perez987 I tried it, but it might not be the right fit for me. After using SSDT-S3, my machine restarts automatically after going to sleep. I’ll look for some SSDT-S3 files that might work for my setup and give it another shot. Thanks a lot for your help!

Oh, and just to mention, I’ve tried different combinations of the methods from the post, but none of them managed to get both sleep and Bluetooth working at the same time. Could be because my system version is 14.6.1 too. I’m going to try different system versions. Thanks a lot for your help!

**Comment by perez987** [comment]
@deanls1 
Forget SSDT-S3. Not working for me neither. 
It's **2. SSDT-GPRW**, this is the method that works, skip the others.

**Comment by deanls1** [comment]
@perez987 

Hi, sorry for the late feedback! I tried the SSDT-GPRW you mentioned, but it still didn't work for me. I also tested a bunch of different system versions, and S3 sleep just wouldn't cooperate. Today, I thought about dynamically unloading Bluetooth with these commands:

```bash
sudo launchctl bootout system/com.apple.bluetoothd
sudo launchctl bootstrap system /System/Library/LaunchDaemons/com.apple.bluetoothd.plist
```

Turns out, these commands worked! And when I pair them with `sleepwatcher`, I can finally get S3 sleep to work and turn Bluetooth back on when I wake up.

**Comment by perez987** [comment]
@deanls1 
Congratulations. You have found another way to solve this issue. Can you explain more deeply how have you done it? When you have time, of course. Sleepwatcher needed but what about configuration or shell scripts and the rest? 
Thank you.

**Comment by perez987** [comment]
@deanls1 
I've tried to do what you say, using sleepwatcher to have a sleep script and an wake script, to disable/re-enable Bluetooth. No success.
I can't get to do it right.
When you can, please tell me how you did it.
Thank you very much.

**Comment by perez987** [comment]
@deanls1 
Closing for now due to lack of response.
